### ISSUE TITLE
[JENKINS-23917] Protocol deadlock while uploading artifacts from ppc64

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#23280]
<p>I've encountered an ssh2 channel protocol issue when a ppc64 slave communicates with an x64 master.</p>

<p>Most operations, like sending build logs, work fine. When the time comes to upload artifacts at the end of the build the build stalls  indefinitely at:</p>

<pre>
Archiving artifacts
</pre>

<p>If I get stack dumps of slave and master using jstack, I see the master waiting to read from the slave:</p>

<pre>
<span class="code-quote">"Channel reader thread: Fedora16-ppc64-Power7-osuosl-karman"</span> prio=10 tid=0x00000000038c2800 nid=0x6de7 in <span class="code-object">Object</span>.wait() [0x00007f825ef8b000]
   java.lang.<span class="code-object">Thread</span>.State: WAITING (on object monitor)
at java.lang.<span class="code-object">Object</span>.wait(Native Method)
- waiting on <0x00000000bf5802e0> (a com.trilead.ssh2.channel.Channel)
at java.lang.<span class="code-object">Object</span>.wait(<span class="code-object">Object</span>.java:502)
at com.trilead.ssh2.channel.FifoBuffer.read(FifoBuffer.java:212)
- locked <0x00000000bf5802e0> (a com.trilead.ssh2.channel.Channel)
at com.trilead.ssh2.channel.Channel$Output.read(Channel.java:127)
at com.trilead.ssh2.channel.ChannelManager.getChannelData(ChannelManager.java:946)
- locked <0x00000000bf5802e0> (a com.trilead.ssh2.channel.Channel)
at com.trilead.ssh2.channel.ChannelInputStream.read(ChannelInputStream.java:58)
at com.trilead.ssh2.channel.ChannelInputStream.read(ChannelInputStream.java:79)
at hudson.remoting.FlightRecorderInputStream.read(FlightRecorderInputStream.java:82)
at hudson.remoting.ChunkedInputStream.readHeader(ChunkedInputStream.java:67)
at hudson.remoting.ChunkedInputStream.readUntilBreak(ChunkedInputStream.java:93)
at hudson.remoting.ChunkedCommandTransport.readBlock(ChunkedCommandTransport.java:33)
at hudson.remoting.AbstractSynchronousByteArrayCommandTransport.read(AbstractSynchronousByteArrayCommandTransport.java:34)
at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)
</pre>

<p>and the slave is waiting for data from the master:</p>

<pre>
<span class="code-quote">"Channel reader thread: channel"</span> prio=10 tid=0x00000fff940fedd0 nid=0x558e runnable [0x00000fff6dc6d000]
   java.lang.<span class="code-object">Thread</span>.State: RUNNABLE
at java.io.FileInputStream.readBytes(Native Method)
at java.io.FileInputStream.read(FileInputStream.java:236)
at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
at java.io.BufferedInputStream.read(BufferedInputStream.java:254)
- locked <0x00000fff78ba9f98> (a java.io.BufferedInputStream)
at hudson.remoting.FlightRecorderInputStream.read(FlightRecorderInputStream.java:82)
at hudson.remoting.ChunkedInputStream.readHeader(ChunkedInputStream.java:67)
at hudson.remoting.ChunkedInputStream.readUntilBreak(ChunkedInputStream.java:93)
at hudson.remoting.ChunkedCommandTransport.readBlock(ChunkedCommandTransport.java:33)
at hudson.remoting.AbstractSynchronousByteArrayCommandTransport.read(AbstractSynchronousByteArrayCommandTransport.java:34)
at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)
</pre>

<p>of course I can't get those dumps at exactly the same moment, even if that were meaningful with network latencies and buffering, but repeated runs never show any other state for either thread.</p>

<p>tshark shows that there's some SSH chatter going on:</p>

<pre>
  0.000000 SLAVE -> MASTER SSH 126 Encrypted response packet len=60
  0.176121 MASTER -> SLAVE SSH 94 Encrypted request packet len=28
  0.176151 SLAVE -> MASTER TCP 66 ssh > 37501 [ACK] Seq=61 Ack=29 Win=707 Len=0 TSval=4141397874 TSecr=2808266826
</pre>

<p>but it should well be low level ssh keepalives or similar, as it's at precise 5 second intervals with nothing much else happening. There are three master->slave ssh connections, so it's not guaranteed that it's even the one associated with the stuck channel.</p>

<p>My first thought is endianness.</p>

<p>I don't really know how to begin debugging this issue, though.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ringerc">ringerc</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23917" target="_blank">Protocol deadlock while uploading artifacts from ppc64</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: core, remoting
<li><b>label(s)</b>: remoting, slave, ssh, unix
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2023-02-03T20:20:49+00:00
<li><b>votes</b>: 2
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I've encountered an ssh2 channel protocol issue when a ppc64 slave communicates with an x64 master.</p>

<p>Most operations, like sending build logs, work fine. When the time comes to upload artifacts at the end of the build the build stalls  indefinitely at:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Archiving artifacts
</pre>
</div></div>

<p>If I get stack dumps of slave and master using jstack, I see the master waiting to read from the slave:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-quote">"Channel reader thread: Fedora16-ppc64-Power7-osuosl-karman"</span> prio=10 tid=0x00000000038c2800 nid=0x6de7 in <span class="code-object">Object</span>.wait() [0x00007f825ef8b000]
   java.lang.<span class="code-object">Thread</span>.State: WAITING (on object monitor)
        at java.lang.<span class="code-object">Object</span>.wait(Native Method)
        - waiting on &lt;0x00000000bf5802e0&gt; (a com.trilead.ssh2.channel.Channel)
        at java.lang.<span class="code-object">Object</span>.wait(<span class="code-object">Object</span>.java:502)
        at com.trilead.ssh2.channel.FifoBuffer.read(FifoBuffer.java:212)
        - locked &lt;0x00000000bf5802e0&gt; (a com.trilead.ssh2.channel.Channel)
        at com.trilead.ssh2.channel.Channel$Output.read(Channel.java:127)
        at com.trilead.ssh2.channel.ChannelManager.getChannelData(ChannelManager.java:946)
        - locked &lt;0x00000000bf5802e0&gt; (a com.trilead.ssh2.channel.Channel)
        at com.trilead.ssh2.channel.ChannelInputStream.read(ChannelInputStream.java:58)
        at com.trilead.ssh2.channel.ChannelInputStream.read(ChannelInputStream.java:79)
        at hudson.remoting.FlightRecorderInputStream.read(FlightRecorderInputStream.java:82)
        at hudson.remoting.ChunkedInputStream.readHeader(ChunkedInputStream.java:67)
        at hudson.remoting.ChunkedInputStream.readUntilBreak(ChunkedInputStream.java:93)
        at hudson.remoting.ChunkedCommandTransport.readBlock(ChunkedCommandTransport.java:33)
        at hudson.remoting.AbstractSynchronousByteArrayCommandTransport.read(AbstractSynchronousByteArrayCommandTransport.java:34)
        at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)
</pre>
</div></div>

<p>and the slave is waiting for data from the master:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-quote">"Channel reader thread: channel"</span> prio=10 tid=0x00000fff940fedd0 nid=0x558e runnable [0x00000fff6dc6d000]
   java.lang.<span class="code-object">Thread</span>.State: RUNNABLE
        at java.io.FileInputStream.readBytes(Native Method)
        at java.io.FileInputStream.read(FileInputStream.java:236)
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
        at java.io.BufferedInputStream.read(BufferedInputStream.java:254)
        - locked &lt;0x00000fff78ba9f98&gt; (a java.io.BufferedInputStream)
        at hudson.remoting.FlightRecorderInputStream.read(FlightRecorderInputStream.java:82)
        at hudson.remoting.ChunkedInputStream.readHeader(ChunkedInputStream.java:67)
        at hudson.remoting.ChunkedInputStream.readUntilBreak(ChunkedInputStream.java:93)
        at hudson.remoting.ChunkedCommandTransport.readBlock(ChunkedCommandTransport.java:33)
        at hudson.remoting.AbstractSynchronousByteArrayCommandTransport.read(AbstractSynchronousByteArrayCommandTransport.java:34)
        at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)
</pre>
</div></div>

<p>of course I can't get those dumps at exactly the same moment, even if that were meaningful with network latencies and buffering, but repeated runs never show any other state for either thread.</p>

<p>tshark shows that there's some SSH chatter going on:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">  0.000000 SLAVE -&gt; MASTER SSH 126 Encrypted response packet len=60
  0.176121 MASTER -&gt; SLAVE SSH 94 Encrypted request packet len=28
  0.176151 SLAVE -&gt; MASTER TCP 66 ssh &gt; 37501 [ACK] Seq=61 Ack=29 Win=707 Len=0 TSval=4141397874 TSecr=2808266826
</pre>
</div></div>

<p>but it should well be low level ssh keepalives or similar, as it's at precise 5 second intervals with nothing much else happening. There are three master-&gt;slave ssh connections, so it's not guaranteed that it's even the one associated with the stuck channel.</p>

<p>My first thought is endianness.</p>

<p>I don't really know how to begin debugging this issue, though.</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins master ver. 1.574-SNAPSHOT (git 17d9093) on x64 Debian 7 with:<br/>
java version &quot;1.6.0_32&quot;<br/>
OpenJDK Runtime Environment (IcedTea6 1.13.4) (6b32-1.13.4-1~deb7u1)<br/>
OpenJDK 64-Bit Server VM (build 23.25-b01, mixed mode)<br/>
Jenkins slave on POWER7 ppc64 Fedora 16 (an OSU OSL community machine), with Java:<br/>
java version &quot;1.6.0_24&quot;<br/>
OpenJDK Runtime Environment (IcedTea6 1.11.1) (fedora-65.1.11.1.fc16-ppc64)<br/>
OpenJDK 64-Bit Zero VM (build 20.0-b12, interpreted mode)
```
</details>
<details><summary><i>6 attachments</i></summary>

- [config.xml](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/26420/config.xml)
- [jenkins-master-idle-stack.txt](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/26423/jenkins-master-idle-stack.txt)
- [jenkins-master-stack.txt](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/26421/jenkins-master-stack.txt)
- [jenkins-slave-stack.txt](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/26422/jenkins-slave-stack.txt)
- [slavelog-from-master.txt](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/26424/slavelog-from-master.txt)
- [slavelog-from-slave.txt](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/26425/slavelog-from-slave.txt)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> ringerc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23917?focusedId=206223&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206223">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm going to re-test with a simplified build and a single executor configured.</p>
</pre>
</details>

</ul>
</details>

<p>I'm going to re-test with a simplified build and a single executor configured.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> ringerc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23917?focusedId=206254&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206254">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've been able to reproduce this with a trivial job and after limiting the node to a single executor. It is not consistently reproducible, it's somewhat random. I suspect it's dependent on the input being archived (10MB of randomly generated data), but it could also be just plain random.</p>

<p>I'll attach the config.xml and thread dumps.</p>
</pre>
</details>

</ul>
</details>

<p>I've been able to reproduce this with a trivial job and after limiting the node to a single executor. It is not consistently reproducible, it's somewhat random. I suspect it's dependent on the input being archived (10MB of randomly generated data), but it could also be just plain random.</p>

<p>I'll attach the config.xml and thread dumps.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> ringerc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23917?focusedId=206255&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206255">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Build log is:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Started by user Craig Ringer
[EnvInject] - Loading node environment variables.
Building remotely on fedora16-ppc64-Power7-osuosl-karman (ppc64 fedora16 ppc linux fedora) in workspace /home/jenkins/workspace/ppctest
[ppctest] $ /bin/sh -xe /tmp/hudson9160124340407748260.sh
+ dd <span class="code-keyword">if</span>=/dev/urandom of=dummy.out bs=1M count=10
10+0 records in
10+0 records out
10485760 bytes (10 MB) copied, 0.604488 s, 17.3 MB/s
Archiving artifacts
</pre>
</div></div>

<p>At the time the master stack was taken there was another build running. I'll see if I can capture another once it's idle.</p>
</pre>
</details>

</ul>
</details>

<p>Build log is:</p>


<pre>
Started by user Craig Ringer
[EnvInject] - Loading node environment variables.
Building remotely on fedora16-ppc64-Power7-osuosl-karman (ppc64 fedora16 ppc linux fedora) in workspace /home/jenkins/workspace/ppctest
[ppctest] $ /bin/sh -xe /tmp/hudson9160124340407748260.sh
+ dd <span class="code-keyword">if</span>=/dev/urandom of=dummy.out bs=1M count=10
10+0 records in
10+0 records out
10485760 bytes (10 MB) copied, 0.604488 s, 17.3 MB/s
Archiving artifacts
</pre>

<p>At the time the master stack was taken there was another build running. I'll see if I can capture another once it's idle.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> ringerc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23917?focusedId=206256&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206256">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>If it is helpful, I can provision access to the build worker for anyone interested in this issue.</p>
</pre>
</details>

</ul>
</details>

<p>If it is helpful, I can provision access to the build worker for anyone interested in this issue.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> ringerc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23917?focusedId=206258&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206258">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Attached a jstack for the master at idle except for the stuck connection.</p>
</pre>
</details>

</ul>
</details>

<p>Attached a jstack for the master at idle except for the stuck connection.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> ringerc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23917?focusedId=206281&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206281">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I cancelled the job after &gt;12 hours. Console output was:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Archiving artifacts
ERROR: Failed to archive artifacts: **/dummy.out
java.io.IOException: java.io.IOException: Failed to extract /home/jenkins/workspace/ppctest/transfer of 1 files
	at hudson.FilePath.readFromTar(FilePath.java:2119)
	at hudson.FilePath.copyRecursiveTo(FilePath.java:2031)
	at jenkins.model.StandardArtifactManager.archive(StandardArtifactManager.java:61)
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:183)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:772)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:736)
	at hudson.model.Build$BuildExecution.post2(Build.java:183)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:685)
	at hudson.model.Run.execute(Run.java:1757)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:234)
Caused by: java.io.IOException
	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:177)
	at hudson.util.HeadBufferingStream.read(HeadBufferingStream.java:61)
	at com.jcraft.jzlib.InflaterInputStream.fill(InflaterInputStream.java:175)
	at com.jcraft.jzlib.InflaterInputStream.read(InflaterInputStream.java:106)
	at org.apache.tools.tar.TarBuffer.readBlock(TarBuffer.java:257)
	at org.apache.tools.tar.TarBuffer.readRecord(TarBuffer.java:223)
	at hudson.org.apache.tools.tar.TarInputStream.read(TarInputStream.java:345)
	at java.io.FilterInputStream.read(FilterInputStream.java:107)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1792)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1769)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:1744)
	at hudson.util.IOUtils.copy(IOUtils.java:40)
	at hudson.FilePath.readFromTar(FilePath.java:2109)
	... 12 more

	at hudson.FilePath.copyRecursiveTo(FilePath.java:2038)
	at jenkins.model.StandardArtifactManager.archive(StandardArtifactManager.java:61)
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:183)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:772)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:736)
	at hudson.model.Build$BuildExecution.post2(Build.java:183)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:685)
	at hudson.model.Run.execute(Run.java:1757)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:234)
Caused by: java.util.concurrent.ExecutionException: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.Request$1.get(Request.java:278)
	at hudson.remoting.Request$1.get(Request.java:210)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:59)
	at hudson.FilePath.copyRecursiveTo(FilePath.java:2034)
	... 11 more
Caused by: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.Request.abort(Request.java:299)
	at hudson.remoting.Channel.terminate(Channel.java:802)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:69)
Caused by: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$PeekInputStream.readFully(ObjectInputStream.java:2323)
	at java.io.ObjectInputStream$BlockDataInputStream.readShort(ObjectInputStream.java:2792)
	at java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:800)
	at java.io.ObjectInputStream.&lt;init&gt;(ObjectInputStream.java:298)
	at hudson.remoting.ObjectInputStreamEx.&lt;init&gt;(ObjectInputStreamEx.java:40)
	at hudson.remoting.AbstractSynchronousByteArrayCommandTransport.read(AbstractSynchronousByteArrayCommandTransport.java:34)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)
Build step <span class="code-quote">'Archive the artifacts'</span> changed build result to FAILURE
Finished: FAILURE
</pre>
</div></div>

<p>and interestingly, an artifact was archived, but it is only 176kB of the target file's 10240kB.</p>

<p>I copied the target file and will see if repeating with the same target file always fails. I've also enabled FINEST trace logging of hudson.remoting. and will see if that's informative.</p>
</pre>
</details>

</ul>
</details>

<p>I cancelled the job after >12 hours. Console output was:</p>


<pre>
Archiving artifacts
ERROR: Failed to archive artifacts: **/dummy.out
java.io.IOException: java.io.IOException: Failed to extract /home/jenkins/workspace/ppctest/transfer of 1 files
	at hudson.FilePath.readFromTar(FilePath.java:2119)
	at hudson.FilePath.copyRecursiveTo(FilePath.java:2031)
	at jenkins.model.StandardArtifactManager.archive(StandardArtifactManager.java:61)
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:183)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:772)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:736)
	at hudson.model.Build$BuildExecution.post2(Build.java:183)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:685)
	at hudson.model.Run.execute(Run.java:1757)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:234)
Caused by: java.io.IOException
	at hudson.remoting.FastPipedInputStream.read(FastPipedInputStream.java:177)
	at hudson.util.HeadBufferingStream.read(HeadBufferingStream.java:61)
	at com.jcraft.jzlib.InflaterInputStream.fill(InflaterInputStream.java:175)
	at com.jcraft.jzlib.InflaterInputStream.read(InflaterInputStream.java:106)
	at org.apache.tools.tar.TarBuffer.readBlock(TarBuffer.java:257)
	at org.apache.tools.tar.TarBuffer.readRecord(TarBuffer.java:223)
	at hudson.org.apache.tools.tar.TarInputStream.read(TarInputStream.java:345)
	at java.io.FilterInputStream.read(FilterInputStream.java:107)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1792)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1769)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:1744)
	at hudson.util.IOUtils.copy(IOUtils.java:40)
	at hudson.FilePath.readFromTar(FilePath.java:2109)
	... 12 more

	at hudson.FilePath.copyRecursiveTo(FilePath.java:2038)
	at jenkins.model.StandardArtifactManager.archive(StandardArtifactManager.java:61)
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:183)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:772)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:736)
	at hudson.model.Build$BuildExecution.post2(Build.java:183)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:685)
	at hudson.model.Run.execute(Run.java:1757)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:234)
Caused by: java.util.concurrent.ExecutionException: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.Request$1.get(Request.java:278)
	at hudson.remoting.Request$1.get(Request.java:210)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:59)
	at hudson.FilePath.copyRecursiveTo(FilePath.java:2034)
	... 11 more
Caused by: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.Request.abort(Request.java:299)
	at hudson.remoting.Channel.terminate(Channel.java:802)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:69)
Caused by: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$PeekInputStream.readFully(ObjectInputStream.java:2323)
	at java.io.ObjectInputStream$BlockDataInputStream.readShort(ObjectInputStream.java:2792)
	at java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:800)
	at java.io.ObjectInputStream.<init>(ObjectInputStream.java:298)
	at hudson.remoting.ObjectInputStreamEx.<init>(ObjectInputStreamEx.java:40)
	at hudson.remoting.AbstractSynchronousByteArrayCommandTransport.read(AbstractSynchronousByteArrayCommandTransport.java:34)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)
Build step <span class="code-quote">'Archive the artifacts'</span> changed build result to FAILURE
Finished: FAILURE
</pre>

<p>and interestingly, an artifact was archived, but it is only 176kB of the target file's 10240kB.</p>

<p>I copied the target file and will see if repeating with the same target file always fails. I've also enabled FINEST trace logging of hudson.remoting. and will see if that's informative.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> ringerc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23917?focusedId=206288&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206288">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>A key clue is that slave.jar appears to be dying suddenly (process terminates) every couple of jobs, when running with a 16kb archive file. Unclear if this was happening before with larger archives and I wasn't noticing because jenkins was relaunching the worker.</p>

<p>This seems to happen after successful job completion though.</p>

<p>After reconfiguring the slave launcher with </p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">ulimit -c unlimited &amp;&amp; 
</pre>
</div></div>

<p>as prefix and </p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"> -slaveLog <span class="code-quote">"log-$(date -Iseconds).txt"</span>
</pre>
</div></div>

<p>then rerunning the small 16kb archive job until the slave died (3rd run), I was able to capture logs from both sides.</p>

<p>I'm now going to test base64 encoded streams (in case it's an ssh 8-bit clean issue) and direct tcp/ip.</p>
</pre>
</details>

</ul>
</details>

<p>A key clue is that slave.jar appears to be dying suddenly (process terminates) every couple of jobs, when running with a 16kb archive file. Unclear if this was happening before with larger archives and I wasn't noticing because jenkins was relaunching the worker.</p>

<p>This seems to happen after successful job completion though.</p>

<p>After reconfiguring the slave launcher with </p>


<pre>
ulimit -c unlimited && 
</pre>

<p>as prefix and </p>


<pre>
 -slaveLog <span class="code-quote">"log-$(date -Iseconds).txt"</span>
</pre>

<p>then rerunning the small 16kb archive job until the slave died (3rd run), I was able to capture logs from both sides.</p>

<p>I'm now going to test base64 encoded streams (in case it's an ssh 8-bit clean issue) and direct tcp/ip.</p>
